test/test_nestedtensor.py

```diff
@@ -6022,6 +6022,67 @@ def test_narrow(self, device):
                 nt.values()[nt.offsets()[i] : (nt.offsets()[i] + nt.lengths()[i])],
             )
 
+    @skipIfTorchDynamo("Test compiles internally")
+    @skipCUDAIf(not SM70OrLater, "GPU capability is < SM70")
+    @torch._dynamo.utils.disable_cache_limit()
+    @dtypes(torch.float32)
+    @parametrize("env", ["eager", "compile", "compile_dynamic"])
+    def test_narrow_on_batch_dim(self, device, dtype, env):
```

Comment:
I wonder how much overlap there is with this test and the narrow OpInfo one. 
Like could the "first few", "middle", and "last" batch items be formulated as sample inputs such that we can have two smaller tests that do more narrow things, e.g. `test_narrow_on_batch_dim_input_validation` and `test_narrow_on_batch_dim_narrow_of_narrow`.

Reply:
> Like could the "first few", "middle", and "last" batch items be formulated as sample inputs?

Definitely and I should do that :) will fix

> And then we could reformalate this one to two smaller tests that do more specific things, e.g. test_narrow_on_batch_dim_input_validation and test_narrow_on_batch_dim_narrow_of_narrow.

yeah these are harder to test with OpInfo so I think it makes sense to break them out

Reply:
oh I just realized the compile tests for narrow-on-narrow weren't actually compiling :p

changing them to actually compile fails them; investigating

Reply:
okay i tracked this down to incorrect clamping. old logic was clamping (start, end) on inner values dim space, but we want to clamp on outer batch dim space. This fixed the data-dependent guard errors.

There's still some work to be done for non-contiguous NJTs apparently, but I've been deprioritizing those in general so I'll just land this as-is